### ISSUE TITLE
Build age gate for user accounts (teach & tools)

### ISSUE THREAD

**Issue by xmatthewx** [MozillaFoundation/plan#525]
#### Goal

Design and build an age verification feature for new sign-ups on Webmaker to comply with COPPA.

#### This heartbeat

 1. gather requirements
 1. gather benchmarks
 1. design
 1. legal review

#### Next heartbeat

 1. build 
  - Android front-end
  - id.webmaker.org front-end (MLN)
  - backend (flag pre- post-verification)
  - redirect or block signups on popcorn, appmaker, goggles
 1. track impact on sign-ups
 1. iterate 

#### RACI

Driver: @xmatthewx, @hannahkane 
Decision: @simonwex
Design: @xmatthewx 
Dev: 
Services: @jbuck 
Metrics: @adamlofting

cc @hannahkane  @davidascher

**Comment by simonwex** [comment]
Devs were hard at work getting api.webmaker.org ready for the load of the snippet. @xmatthewx hurried along some design, but we still have more to do. Bumping to next HB.

**Comment by simonwex** [comment]
@cassiemc what's the best way to get designer-eyes on mozilla/id.webmaker.org/issues/362

**Comment by cassiemc** [comment]
@simonwex This is the way :) @hannahkane – can mozilla/id.webmaker.org#362 fit into @sabrinang's cycle next heartbeat? Shouldn't take longer than a day or two since we can use the UX from webmaker.

**Comment by xmatthewx** [comment]
Draft design is in legal review right now. Once that is approved, we can hand it over for UI translation.

**Comment by cassiemc** [comment]
Perfect, thanks @xmatthewx.

**Comment by hannahkane** [comment]
Is it possible to find someone else to do this? Check out the Design work in the next Teach heartbeat: https://github.com/mozilla/teach.webmaker.org/issues?q=is%3Aopen+milestone%3A%22August+28%2C+2015%22+label%3Adesign

That includes a two-day sprint on the Directory wireframes, and that *doesn't* include whatever remaining Thimble promo work there will be to do.  My sense is that taking out a day or two out for this would require us to remove stuff from the heartbeat.

**Comment by xmatthewx** [comment]
@vazquez could continue this work

**Comment by cassiemc** [comment]
Fab, that is ideal, and would likely take much less time since he's familiar with the UX. Thanks @xmatthewx!

**Comment by simonwex** [comment]
@xmatthewx do you have time to drive this next heartbeat? I didn't have time this one.

**Comment by xmatthewx** [comment]
Yes, I can drive it. Design and legal review are complete. I *think* we have the resources to land this in Webmaker.

**Comment by simonwex** [comment]
@xmatthewx, I should have moved this over to you in assignment. I bet this didn't end up on your radar properly.

**Comment by xmatthewx** [comment]
I'm going to press pause on this, pending planning that will completed during our work in NYC. Moving issue to the backlog.

**Comment by hannahkane** [comment]
Is there a plan to revive this ticket soon?

**Comment by xmatthewx** [comment]
I bumped it the backlog. I don't plan to revive it until we have clarity on 2016 plans.

**Comment by hannahkane** [comment]
Making sure I understand - the age gate is needed for anything that uses is, right? Including Thimble, Goggles, and Teach?

**Comment by xmatthewx** [comment]
Yes. I believe it was particularly urgent for Webmaker given it's marketing in the play store that might attract youth. We do have UX/UI done, that could fairly easily be translated to login for those desktop tools.

**Comment by hannahkane** [comment]
This really needs to happen. Can we take it out of the backlog and move it to an upcoming milestone? @xmatthewx @simonwex

**Comment by xmatthewx** [comment]
Sure thing. We have design and thinking and copy done.

**Comment by xmatthewx** [comment]
I'd suggest we implement on desktop/teach site first, rather than within the android app.

**Comment by simonwex** [comment]
I agree with @xmatthewx. @hannahkane let's chat when we're back about the who and timing.

**Comment by simonwex** [comment]
@xmatthewx could we re-write this ticket for the HB deliverable?

**Comment by xmatthewx** [comment]
Yes. That's a great idea. I'll probably close this monster and open a new one in collab with @hannahkane

**Comment by OpenMatt** [comment]
@hannahkane is interested in this.

**Comment by xmatthewx** [comment]
tracking in new ticket: https://github.com/MozillaFoundation/plan/issues/525
